pkg/apis/workflow/v1alpha1/container_set_template_types.go

```diff
@@ -42,6 +46,62 @@ func (in *ContainerSetTemplate) HasSequencedContainers() bool {
 	return false
 }
 
+// Validate checks if the ContainerSetTemplate is valid
+func (in *ContainerSetTemplate) Validate(tmpl *Template) error {
```

Comment:
this seems upside down to me - this should not take any parameters - any validated of the template should be done in `Template.Validate()`

Reply:
Thanks for the review @alexec!

I just pushed the latest commit to move the validation out.